api_endpoints.yaml

```diff
@@ -0,0 +1,1732 @@
+      security:
+        - userAuth: []
+      tags:
+        - "schedule"
+      summary: ""
+      description: ""
+      operationId: ""
+      produces:
+      - "application/json"
+      responses:
+        "200":
+          description: "Successful operation"
+  /schedule/{id}:
+    get:
+      security:
+        - userAuth: []
+      tags:
+        - "schedule"
+      summary: ""
+      description: ""
+      operationId: ""
+      produces:
+      - "application/json"
+      parameters:
+        - name: "id"
+          in: "path"
+          description: "ID needed to retrieve schedule"
+          required: true
+          type: "integer"
+      responses:
+        "200":
+          description: "Successful operation"
+        "400":
+          description: "Invalid id supplied"
+        "404":
+          description: "Object not found"  
+    delete:
+      security:
+        - userAuth: []
+      tags:
+        - "schedule"
+      summary: ""
+      description: ""
+      operationId: ""
+      produces:
+      - "application/json"
+      parameters:
+        - name: "id"
+          in: "path"
+          description: "ID needed to delete schedule"
+          required: true
+          type: "integer"
+      responses:
+        "200":
+          description: "Successful operation"
+        "400":
+          description: "Invalid id supplied"
+        "404":
+          description: "Object not found"  
+  /schedule/sessions:
+    post:
+      security:
+        - userAuth: []
+      tags:
+        - "schedule"
+      summary: ""
+      description: "Details TBD"
+      operationId: ""
+      produces:
+      - "application/json"
+      responses:
+        "200":
+          description: "Successful operation"
+          
+          
+securityDefinitions:
+  userAuth:
+    type: basic
+
+securtiy:
+  - userAuth: []
+  
+definitions:
+  Profile:
+    type: "object"
+    properties:
+      profile_id:
+        type: "integer"
+        format: "int64"
+      email:
+        type: "string"
+      name:
+        type: "string"
+      auth0_id:
+        type: "string"
+      role:
+        type: "string"
+        description: "Role status"
+        enum:
+        - "super_admin"
+        - "admin"
+        - "instructor"
+        - "parent"
+        - "user"
+      avatarUrl:
+        type: "string"
+      pending:
+        type: "string"
+  User:
+    type: "object"
+    properties:
+      firstName:
+        type: "string"
+      lastName:
+        type: "string"
+      email:
+        type: "string"
+      role_id:
+        type: "integer"
+        description: "Role status"
+      avatarUrl:
+        type: "string"
+  Parent:
+    description: "Details TBD"
+  Children:
+    type: "object"
+    properties:
+      child_id:
+        type: "integer"
+      profile_id: 
+        type: "integer"
+      username: 
+        type: "string"
+        example: "Tommy"
+      age: 
+        type: "integer"
+      parent_id:
+        type: "integer"
```

Comment:
For the instructor definition, I believe these are the properties that will be used 

```
{
  instructor_id: INCREMENT (primary key, auto-increments, generated by database),
  instructor_name: STRING (required),
  rating: INTEGER (required),
  availability: STRING (optional),
  bio: STRING (required),
  profile_id: INTEGER (required, foreign key),
  status: STRING (required, default: 'pending'),
  approved_by: INTEGER (optional, foreign key, default: null, references admin_id),
}
```